doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -1,102 +1,126 @@
-=================================
-Term rewriting and simplification
-=================================
+=========================
+Reasoning with equalities
+=========================
 
-.. _rewritingexpressions:
+There are multiple notions of :gdef:`equality` in Coq:
 
-Rewriting expressions
----------------------
+- :gdef:`Leibniz equality` is the standard
+  way to define equality in Coq and the Calculus of Inductive Constructions,
+  which is in terms of a binary relation, i.e. a binary function that returns
+  a `Prop`.  The standard library
+  defines `eq` similar to this:
 
-These tactics use the equality :g:`eq:forall A:Type, A->A->Prop` defined in
-file ``Logic.v`` (see :ref:`coq-library-logic`). The notation for :g:`eq T t u` is
-simply :g:`t=u` dropping the implicit type of :g:`t` and :g:`u`.
+   .. coqdoc::
 
-.. tacn:: rewrite @term
-   :name: rewrite
+      Inductive eq {A : Type} (x : A) : A -> Prop := eq_refl : eq x x.
 
-   This tactic applies to any goal. The type of :token:`term` must have the form
+  The notation `x = y` represents the term `eq x y`.  The notation `x = y :> A`
+  gives the type of x and y explicitly.
 
-   ``forall (x``:sub:`1` ``:A``:sub:`1` ``) ... (x``:sub:`n` ``:A``:sub:`n` ``), eq term``:sub:`1` ``term``:sub:`2` ``.``
+- :gdef:`Setoid equality <setoid equality>` defines equality in terms of an equivalence
+  relation.  A :gdef:`setoid` is a set that is equipped with an equivalence relation
+  (see https://en.wikipedia.org/wiki/Setoid).  These are needed to form a :gdef:`quotient set`
+  or :gdef:`quotient`
+  (see https://en.wikipedia.org/wiki/Equivalence_Class).  In Coq, users generally work
+  with setoids rather than constructing quotients, for which there is no specific support.
 
-   where :g:`eq` is the Leibniz equality or a registered setoid equality.
+- :gdef:`Definitional equality <definitional equality>` is equality based on the
+  :ref:`conversion rules <Conversion-rules>`, which Coq can determine automatically.
+  When two terms are definitionally equal, Coq knows it can
+  replace one with the other, such as with :tacn:`change` `X with Y`, among many
+  other advantages.  ":term:`Convertible <convertible>`" is another way of saying that
```

Comment:
We could also add a reference from the place where "convertible" is defined to here. I would simply change the `or *equivalent*` phrase over there to ``` or :term:`definitionally equal <definitional equality>` ```.